en/docs/SetupAndInstall/UpgradingWSO2APIManager/upgrading-from-250-to-300.md

```diff
@@ -3,350 +3,1676 @@
+            ALTER COLUMN LAST_ACCESSED TYPE TIMESTAMP,
+            ALTER COLUMN CREATED_TIME TYPE TIMESTAMP,
+            ALTER COLUMN UPDATED_TIME TYPE TIMESTAMP;
+
+        ALTER TABLE AM_API_LC_EVENT
+            ALTER COLUMN EVENT_DATE TYPE TIMESTAMP,
+            ALTER COLUMN EVENT_DATE SET NOT NULL;
+
+        ALTER TABLE AM_API_COMMENTS
+            ALTER COLUMN DATE_COMMENTED TYPE TIMESTAMP,
+            ALTER COLUMN DATE_COMMENTED SET NOT NULL;
+
+        DROP TABLE IF EXISTS AM_ALERT_TYPES;
+
+        DROP SEQUENCE IF EXISTS  AM_ALERT_TYPES_SEQ;
+
+        CREATE SEQUENCE AM_ALERT_TYPES_SEQ START WITH 1 INCREMENT BY 1;
+        CREATE TABLE IF NOT EXISTS AM_ALERT_TYPES (
+                    ALERT_TYPE_ID INTEGER DEFAULT NEXTVAL('am_alert_types_seq'),
+                    ALERT_TYPE_NAME VARCHAR(255) NOT NULL ,
+                STAKE_HOLDER VARCHAR(100) NOT NULL,           
+                    PRIMARY KEY (ALERT_TYPE_ID)
+        );
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('AbnormalResponseTime', 'publisher');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('AbnormalBackendTime', 'publisher');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('AbnormalRequestsPerMin', 'subscriber');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('AbnormalRequestPattern', 'subscriber');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('UnusualIPAccess', 'subscriber');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('FrequentTierLimitHitting', 'subscriber');
+
+        INSERT INTO AM_ALERT_TYPES (ALERT_TYPE_NAME, STAKE_HOLDER) VALUES ('ApiHealthMonitor', 'publisher');
+
+        ALTER TABLE AM_CERTIFICATE_METADATA DROP CONSTRAINT IF EXISTS END_POINT_CONSTRAINT;
+
+        CREATE SEQUENCE AM_SYSTEM_APP_SEQUENCE START WITH 1 INCREMENT BY 1 ;
+        CREATE TABLE IF NOT EXISTS AM_SYSTEM_APPS (
+                    ID INTEGER DEFAULT nextval('am_system_app_sequence'),
+                    NAME VARCHAR(50) NOT NULL,
+                    CONSUMER_KEY VARCHAR(512) NOT NULL,
+                    CONSUMER_SECRET VARCHAR(512) NOT NULL,
+                    CREATED_TIME TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6),
+                    PRIMARY KEY (ID)
+        );
+
+        CREATE TABLE IF NOT EXISTS AM_API_CLIENT_CERTIFICATE (
+        TENANT_ID INTEGER NOT NULL,
+        ALIAS VARCHAR(45) NOT NULL,
+        API_ID INTEGER NOT NULL,
+        CERTIFICATE BYTEA NOT NULL,
+        REMOVED BOOLEAN NOT NULL DEFAULT 0,
+        TIER_NAME VARCHAR (512),
+        FOREIGN KEY (API_ID) REFERENCES AM_API (API_ID) ON DELETE CASCADE ON UPDATE CASCADE,
+        PRIMARY KEY (ALIAS,TENANT_ID, REMOVED)
+        );
+
+        ALTER TABLE AM_POLICY_SUBSCRIPTION ADD MONETIZATION_PLAN VARCHAR(25) NULL DEFAULT NULL,
+        ADD FIXED_RATE VARCHAR(15) NULL DEFAULT NULL, 
+        ADD BILLING_CY
```

Comment:
We cannot add the jar directly to lib folder in Analytics servers. Lets modify the step as below.

Do the following to integrate third party products when configuring databases. 
WSO2 SI is a OSGi-based product. Therefore, when you integrate third party products such as Oracle with WSO2 API-M Analytics, you need to check whether the libraries you need to add are OSGi based. If they are not, you need to convert them to OSGi bundles before adding them to the <API-M_ANALYTICS_HOME>/lib directory. To convert the jar files to OSGi bundles, follow the steps below. 
1. Download the non-OSGi jar for the required third party product, and save it in a preferred directory in your machine.
2. Go to the <API-M_ANALYTICS_HOME>/bin directory. Run the command given below, to generate the converted file in the <API-M_ANALYTICS_HOME>/lib directory.
`./jartobundle.sh <PATH_TO_NON-OSGi_JAR> ../lib`

Reply:
Fixed